nanome/util/string_builder.py

```diff
@@ -5,8 +5,11 @@ def __init__(self):
     def append(self, s):
         self.los.append(str(s))
 
-    def to_string(self):
-        joined = "".join(self.los)
+    def append_string(self, s):
+        self.los.append(s)
+
+    def to_string(self, joiner = ""):
```

Comment:
having a "joiner" character means that some uses of the string builder won't need to append as many times.